challenge/dns01/nameserver.go

```diff
@@ -229,15 +231,39 @@ func dnsQuery(fqdn string, rtype uint16, nameservers []string, recursive bool) (
 	m := createDNSMsg(fqdn, rtype, recursive)
 
 	var in *dns.Msg
-	var err error
+	var errG error
 
 	for _, ns := range nameservers {
-		in, err = sendDNSQuery(m, ns)
-		if err == nil && len(in.Answer) > 0 {
+		bo := backoff.NewExponentialBackOff()
+		bo.Multiplier = 1.2
+		bo.InitialInterval = dnsTimeout
+		bo.MaxInterval = 2 * bo.InitialInterval
+		bo.MaxElapsedTime = 4 * bo.InitialInterval
```

Comment:
This sets the max. wait time to 2 minutes if I invoke lego with `--dns-timeout=30`, doesn't it? Should `bo.InitialInterval`  be `dnsTimeout/4` instead?

Reply:
the initial interval must be the same as the timeout use by the DNS client.

Reply:
> The initial interval must be the same as the timeout use by the DNS client

I see. Then the documentation of the `--dns-timeout` flag needs to be modified to something like "this is the minimum timeout given to resolve SOA queries; on network errors the actual timeout might be considerably longer".

Alternatively, we can reduce the default timeout down to something like 2 or 3s to approximate the previous behaviour (on that note, this should work in *most* cases, because DNS *usually* resolves fast enough). To accommodate #1008, we could also bump `bo.MaxInterval` and `bo.MaxElapsedTime` to `10*initialInterval` and `20*initialInterval` respectively.

Reply:
I can reduce the default timeout.

But I thank that multiply by 10 or 20 is too huge: the dnsQuery is used by several functions in a loop (by example a loop on ns) and this can become very very long.

Reply:
Oh, I've missed the context; I thought this was part of the `fetchSoaByFqdn` methods. Sorry for the confusion on my part.